source/Octopus.Tentacle.Tests.Integration/ScriptServiceTests.cs

```diff
@@ -0,0 +1,125 @@
+                .ExecuteScript(windowsScript, nixScript, CancellationToken);
+            DumpLog(scriptStatusResponse);
+
+            scriptStatusResponse.State.Should().Be(ProcessState.Complete);
+            scriptStatusResponse.ExitCode.Should().Be(0);
+            scriptStatusResponse.Logs.Select(x => x.Text).Should().Contain("The answer is 42");
+        }
+
+        [Test]
+        [TestCaseSource(typeof(TentacleTypesToTest))]
+        public async Task RunScriptWithErrors(TentacleType tentacleType)
+        {
+            var windowsScript = @"
+                Write-Host ""This is the start of the script""
+                Start-Sleep -Seconds 3
+                throw ""Whoopsy Daisy!""
+                Write-Host ""This is the end of the script""";
+
+            var nixScript = @"
+                echo This is the start of the script
+                sleep 3
+                echo ""Whoopsy Daisy!""
+                exit 1
+                echo This is the end of the script";
+
+            using var clientAndTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .Build(CancellationToken);
+
+            var scriptStatusResponse = await new ScriptExecutionOrchestrator(clientAndTentacle.TentacleClient)
+                .ExecuteScript(windowsScript, nixScript, CancellationToken);
+            DumpLog(scriptStatusResponse);
+
+            scriptStatusResponse.State.Should().Be(ProcessState.Complete);
+            scriptStatusResponse.ExitCode.Should().NotBe(0);
+            scriptStatusResponse.Logs.Select(x => x.Text).Should().Contain("Whoopsy Daisy!");
+            scriptStatusResponse.Logs.Select(x => x.Text).Should().NotContain("This is the end of the script");
+        }
+
+        [Test]
+        [TestCaseSource(typeof(TentacleTypesToTest))]
+        public async Task CancelScript(TentacleType tentacleType)
+        {
+            var windowsScript = @"Write-Host ""This is the start of the script""
+                                & ping.exe 127.0.0.1 -n 100
+                                Write-Host ""This is the end of the script""";
+
+            var nixScript = @"echo This is the start of the script
+                              ping 127.0.0.1 -c 100
+                              echo This is the end of the script";
+
+            using var clientAndTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .Build(CancellationToken);
+
+            var scriptExecutor = new ScriptExecutionOrchestrator(clientAndTentacle.TentacleClient);
+            var ticket = scriptExecutor.StartScript(windowsScript, nixScript, CancellationToken);
+            // Possible Tentacle BUG: If we just observe until the first output is received then sometimes the script will fail to Cancel
+            await scriptExecutor.ObserverUntilScriptOutputReceived(ticket, "This is the start of the script", CancellationToken);
+
+            Console.WriteLine("Canceling");
```

Comment:
Consider using serilog (can be made with `SerilogLoggerBuilder`).

That way the messages will have timestamps as well as information about what test it is being run in.